### ISSUE TITLE
NoMethodError: undefined method `title' for #<Nokogiri::XML::Document:0x007fe6605bf290>

### ISSUE THREAD

**Issue by filterfish** [sparklemotion/nokogiri#1599]
I'm trying to parse the following document: https://lists.debian.org/debian-devel-announce/2017/02/msg00001.html but I reliably get a `NoMethodError` exception.

This code fragment causes the exception:

```ruby
@document.title
```

Whereas this works as you'd exepct:

```ruby
@document.search('//title').inner_text
```

This is not a big one for me as I've got a workaround but I thought you might want to know! I've attached the file (with a `.txt` suffix) in case the URI goes away.

[debian-devel-announce.txt](https://github.com/sparklemotion/nokogiri/files/753750/debian-devel-announce.txt)

**Comment by larskanis** [comment]
You must open your file as a Nokogiri::HTML::Document in order to use the title method.

**Comment by flavorjones** [comment]
@filterfish can you please let us know whether Lars's advice helps?

**Comment by filterfish** [comment]
Thanks @larskanis that worked a treat!

So does `Nokogiri` auto-detect the document type? The reason I ask it that the code I've currently got works on most documents it's just some that fail.

Um, I think I've just answered my own question: I've just printed the `@document.class` and lo and behold sometimes it's `Nokogiri::HTML::Document` and sometimes it's `Nokogiri::XML::Document`. What's more when the document type is `Nokogiri::XML::Document` it fails.

Closing. Sorry for the noise.
